src/class/hid/hid_device.h

```diff
@@ -380,7 +380,30 @@ static inline bool  tud_hid_gamepad_report(uint8_t report_id, int8_t x, int8_t y
 // HID Generic Input & Output
 // - 1st parameter is report size (mandatory)
 // - 2nd parameter is report id HID_REPORT_ID(n) (optional)
-#define TUD_HID_REPORT_DESC_GENERIC_INOUT(report_size, ...) \
+#if (CFG_TUSB_REPORT_ID_COUNT == 2)
+  #define TUD_HID_REPORT_DESC_GENERIC_INOUT(report_size, ...) \
```

Comment:
please explain what is this template used for ?

Reply:
I found that the macro didn't support two different Report IDs(in / out), so I tried to separate them.

Reply:
For example: If I need a report id(1) for out and another one(2) for in,  I won't get the data throuth this original macro.

Reply:
During designing a usb generic HID communication, I found that the macro didn't support two different Report IDs(in / out), so I tried to separate them.
For example: 
If I need a report id(Report ID: 1) for out reports and another one(Rreport ID: 2) for in reports, I won't get the data through this original macro.

Reply:
if template descriptor does not fit your need, you don't need to use it, just make your own descritpor. These template only cover the most common usage by user. Therefore this should be reverted !